modules/benchmarks/src/main/java/org/apache/ignite/internal/benchmarks/jmh/streamer/JmhStreamerAddDataBenchmark.java

```diff
@@ -0,0 +1,201 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.benchmarks.jmh.streamer;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.logger.NullLogger;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.cache.CacheWriteSynchronizationMode.FULL_SYNC;
+
+/**
+ * DataStreamerImpl.addData(Collection) vs DataStreamerImpl.addData(Key, Value).
+ */
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class JmhStreamerAddDataBenchmark {
+    /** Data amount. */
+    private static final int DATA_AMOUNT = 7000;
+
+    /** Ignite client instance. */
+    private static final String IGNITE_CLIENT_INSTANCE_NAME = "client";
+
+    /** Ignite client cache name. */
+    private static final String IGNITE_CLIENT_CACHE_NAME = "clientCache";
+
+    /**
+     * Create Ignite configuration.
+     *
+     * @return Ignite configuration.
+     */
+    private IgniteConfiguration getConfiguration(String cfgName, boolean isClient) {
+        IgniteConfiguration cfg = new IgniteConfiguration();
+
+        cfg.setGridLogger(new NullLogger());
+
+        cfg.setIgniteInstanceName(cfgName);
+
+        if (isClient) {
+            cfg.setClientMode(true);
+
+            cfg.setCacheConfiguration(defaultCacheConfiguration(IGNITE_CLIENT_CACHE_NAME));
+        }
+        else
+            cfg.setCacheConfiguration(defaultCacheConfiguration("server"));
+
+        return cfg;
+    }
+
+    /**
+     * Create cache configuration.
+     *
+     * @return Cache configuration.
+     */
+    private CacheConfiguration defaultCacheConfiguration(String cacheName) {
+        CacheConfiguration cfg = new CacheConfiguration(cacheName);
+
+        cfg.setAtomicityMode(TRANSACTIONAL);
```

Comment:
let's make `atomicityMode()` method that return `TRANSACTIONAL` to make life easier for the one who wants to extend this test.